ibis_ml/steps/_encode.py

```diff
@@ -352,9 +357,9 @@ def fit_table(self, table: ir.Table, metadata: Metadata) -> None:
                     + self.target_means_[target] * self.smooth
                 ) / (agged[f"{target}_count"] + self.smooth)
 
-            self.encodings_[column] = ibis.memtable(
-                agged.mutate(**target_encodings).drop(target_aggs).to_pyarrow()
-            )
+            encoding_expr = agged.mutate(**target_encodings).drop(target_aggs)
+            self._fit_expr.append(encoding_expr)
+            self.encodings_[column] = ibis.memtable(self._fit_expr[-1].to_pyarrow())
```

Comment:
```suggestion
            encodings_expr = agged.mutate(**target_encodings).drop(target_aggs)
            self._fit_expr.append(encodings_expr)
            self.encodings_[column] = ibis.memtable(encodings_expr.to_pyarrow())
```
Nit: Same, more clear than indexing